content/docs/guides/opentelemetry.md

```diff
@@ -0,0 +1,208 @@
+
+
+``` yaml
+storage:
+  tsdb:
+    # A 10min time window is enough because it can easily absorb retries and network delays.
+    out_of_order_time_window: 10m
+```
+
+The support for OpenTelemetry Metrics ingestion is supported in Prometheus via a feature flag. Run Prometheus with the `otlp-write-receiver` flag enabled as follows:
+
+``` shell
+prometheus --enable-feature=otlp-write-receiver --config=prometheus.yml
+```
+
+Once the feature is successfully enabled, you can see the evidence in the logs.
+
+``` shell
+$ prometheus --enable-feature=otlp-write-receiver
+ts=2023-09-03T04:11:01.456Z caller=main.go:175 level=info msg="Experimental OTLP write receiver enabled"
+```
+
+## Setting up a Python Flask Application
+
+We will use a Flask API application for the demonstration in this guide. The application performs calculator operations and responds to `add`, `subtract`, `multiply` and `divide` operations.
+
+> It is assumed that you have latest version of Python and `pip` setup.
+
+``` shell
+mkdir otel-to-prom
+cd otel-to-prom
+pip install Flask
+```
+
+Create the application.
+
+``` python
+from flask import Flask, jsonify, request
+
+app = Flask(__name__)
+
+@app.route('/add', methods=['POST'])
+def add():
+    data = request.json
+    result = data['num1'] + data['num2']
+    return jsonify({"result": result})
+
+@app.route('/subtract', methods=['POST'])
+def subtract():
+    data = request.json
+    result = data['num1'] - data['num2']
+    return jsonify({"result": result})
+
+@app.route('/multiply', methods=['POST'])
+def multiply():
+    data = request.json
+    result = data['num1'] * data['num2']
+    return jsonify({"result": result})
+
+@app.route('/divide', methods=['POST'])
+def divide():
+    data = request.json
+    if data['num2'] == 0:
+        return jsonify({"error": "Division by zero is not allowed."}), 400
+    result = data['num1'] / data['num2']
+    return jsonify({"result": result})
+
+if __name__ == '__main__':
+    app.run(debug=True)
+
+```
+
+Start the application and send a couple of requests:
+
+``` shell
+$ python app.py
+ * Serving Flask app 'app'
+ * Debug mode: on
+WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
+ * Running on http://127.0.0.1:5000
+Press CTRL+C to quit
+ * Restarting with stat
+ * Debugger is active!
+ * Debugger PIN: 988-101-184
+127.0.0.1 - - [03/Sep/2023 09:48:09] "POST /add HTTP/1.1" 200 -
+```
+
+``` shell
+$ curl -X POST -H "Content-Type: application/json" -d '{"num1": 5, "num2": 100}' http://127.0.0.1:5000/add
+
+{
+  "result": 105
+}
+```
+
+## Auto Instrumentation
+
+Let's add the OpenTelemetry instrumentation now. We will use Otel's Flask instrumentation library which auto-instruments all request handelers.
+
+``` shell
+pip opentelemetry-api opentelemetry-sdk opentelemetry-exporter-otlp opentelemetry-instrumentation-flask
```

Comment:
Would be nice if example included creation of virtualenv. Installing to system python is against best practice. In newer versions of ubuntu, pip install fails and asks to provide `--break-system-packages` arg.